eks/deploy.go

```diff
@@ -57,112 +58,157 @@ func RollOutDeployment(
 	elbv2Svc := elbv2.New(sess)
 	logger.Infof("Successfully authenticated with AWS")
 
-	// Retrieve the ASG object and gather required info we will need later
-	asgInfo, err := getAsgInfo(asgSvc, eksAsgName)
-	if err != nil {
-		return err
+	stateFile := defaultStateFile
+
+	// Retrieve state if one exists or construct a new one
+	state, err := readOrInitializeState(stateFile, ignoreRecoveryFile)
+
+	// If we're in the initial state, gather ASG info and wait for capacity
+	if !state.GatherASGInfoDone {
```

Comment:
I think this if based state machine is fine, but perhaps it would make sense to encapsulate each stage as a method on the state machine struct for better maintainability? E.g.:

```
if !state.ActionDone {
  state.action()
}
```

This should hopefully make it easier to extend with additional state actions in the future.